packages/trpc/server/routers/viewer/teams/inviteMember.handler.ts

```diff
@@ -112,13 +113,35 @@ export const inviteMemberHandler = async ({ ctx, input }: InviteMemberOptions) =
     }
     // inform user of membership by email
     if (input.sendEmailInvitation && ctx?.user?.name && team?.name) {
+      const inviteTeamOptions = {
+        joinLink: `${WEBAPP_URL}/auth/login?callbackUrl=/settings/teams`,
+        isCalcomMember: true,
+      };
+      /**
+       * Here we want to redirect to a differnt place if onboarding has been completed or not. This prevents the flash of going to teams -> Then to onboarding - also show a differnt email template.
+       * This only changes if the user is a CAL user and has not completed onboarding and has no password
+       */
+      if (!invitee.completedOnboarding && !invitee.password && invitee.identityProvider === "CAL") {
+        const token = randomBytes(32).toString("hex");
+        await prisma.verificationToken.create({
+          data: {
+            identifier: input.usernameOrEmail,
+            token,
+            expires: new Date(new Date().setHours(168)), // +1 week
+          },
+        });
+
+        inviteTeamOptions.joinLink = `${WEBAPP_URL}/signup?token=${token}&callbackUrl=/getting-started`;
+        inviteTeamOptions.isCalcomMember = false;
```

Comment:
They ARE a calcomMember here but they havent completed onboarding - so we show them the new user email invite which explains the onboarding flow

<img width="683" alt="CleanShot 2023-05-10 at 09 36 32@2x" src="https://github.com/calcom/cal.com/assets/55134778/37b4720f-50b5-41f3-bc29-c53387f4453b">